tests/backup/backup_psa_test.go

```diff
@@ -28,3 +41,605 @@ var _ = Describe("{DummyPSATestcase}", Label(TestCaseLabelsMap[DummyPSATestcase]
+				storageClassMapping[sc.Name] = scName
+			}
+			log.Infof("Storage class mapping is %v", storageClassMapping)
+
+			// Switching cluster context back to source cluster
+			log.InfoD("Switching cluster context back to source cluster")
+			err = SetSourceKubeConfig()
+			log.FailOnError(err, "Failed to set source kubeconfig")
+
+			namespaceMapping := make(map[string]string)
+			namespaceMapping[restrictedNamespaceInSrcCluster] = privilegeNamespace
+			log.Infof("Namespace mapping is %v", namespaceMapping)
+
+			customRestoreWithNamespaceAndStorageClassMappingToPrivilegedNamespace = fmt.Sprintf("%s-%v-ns-mapping-%v", RestoreNamePrefix, RandomString(3), backupNameWithVolumeAndResourceWithRestrictedLabel)
+			//appContextsToRestore := FilterAppContextsByNamespace(scheduledAppContexts, appNamespaces)
+			err = CreateRestoreWithValidation(ctx, customRestoreWithNamespaceAndStorageClassMappingToPrivilegedNamespace, backupNameWithVolumeAndResourceWithRestrictedLabel, namespaceMapping, storageClassMapping, DestinationClusterName, BackupOrgID, singleScheduledAppContexts)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying %s backup's restore %s creation with namespace mapping %v and storage class mapping %v", backupNameWithVolumeAndResourceWithRestrictedLabel, customRestoreWithNamespaceAndStorageClassMappingToPrivilegedNamespace, namespaceMapping, storageClassMapping))
+			restoreList = append(restoreList, customRestoreWithNamespaceAndStorageClassMappingToPrivilegedNamespace)
+		})
+
+		Step("Take backup of namespace with baseline labele and restore it to a destination cluster on namespace with privileged label", func() {
+			log.InfoD("Taking backup of applications with both volume and resources on namespace with baseline label set")
+			backupNameWithVolumeAndResourceWithBaselineLabel = fmt.Sprintf("%s-%v", BackupNamePrefix, RandomString(10))
+			err = CreateBackupWithValidation(ctx, backupNameWithVolumeAndResourceWithBaselineLabel, SourceClusterName, backupLocationName, backupLocationUID, baselineScheduledAppContexts, nil, BackupOrgID, srcClusterUid, preRuleName, preRuleUid, postRuleName, postRuleUid)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup of namespaces with volume and resources [%s]", backupNameWithVolumeAndResourceWithBaselineLabel))
+			backupList = append(backupList, backupNameWithVolumeAndResourceWithBaselineLabel)
+
+			// Switching cluster context to destination cluster
+			log.InfoD("Switching cluster context to destination cluster")
+			err = SetDestinationKubeConfig()
+			log.FailOnError(err, "Failed to set destination kubeconfig")
+
+			// Create a namespace with privileged PSA type to take restore in that namespace
+			privilegedNamespace := fmt.Sprintf("single-ns-multi-app-%v-privileged-label", RandomString(3))
+			_, err = core.Instance().CreateNamespace(&v1.Namespace{
```

Comment:
Please add a todo here so that we can use CreateNamespaceAndAssignLabels() instead from https://github.com/portworx/torpedo/pull/2579 instead of adding the same code again and again?